python/desc/skycatalogs/objects/base_object.py

```diff
@@ -390,63 +403,57 @@ def get_total_observer_sed(self):
             sed *= mu
         return sed
 
-    def get_flux(self, bandpass, sed=None):
+    def get_flux(self, bandpass, sed=None, mjd=None):
         """
         Return the total object flux integrated over the bandpass
         in photons/sec/cm^2
         Use supplied sed if there is one
         """
 
         if not sed:
-            sed = self.get_total_observer_sed()
+            sed = self.get_total_observer_sed(mjd=mjd)
 
         flux = sed.calculateFlux(bandpass)
 
         return flux
 
-    def get_fluxes(self, bandpasses):
+    def get_fluxes(self, bandpasses, mjd=None):
         # To avoid recomputing sed
-        sed = self.get_total_observer_sed()
+        sed = self.get_total_observer_sed(mjd=mjd)
         return [sed.calculateFlux(b) for b in bandpasses]
 
-    def get_LSST_flux(self, band, sed=None, cache=True):
+    def get_LSST_flux(self, band, sed=None, cache=True, mjd=None):
         if not band in LSST_BANDS:
             return None
-        att = f'lsst_flux_{band}'
-        # Check if it's already an attribute
-        val = getattr(self, att, None)
-        if val is not None:
-            return val
+        if not mjd:
+            att = f'lsst_flux_{band}'
+            # Check if it's already an attribute
+            val = getattr(self, att, None)
+            if val is not None:
+                return val
 
-        if att in self.native_columns:
-            return self.get_native_attribute(att)
+        # For now exclude sn
+        if self.object_type != 'sn':
+            if att in self.native_columns:
+                return self.get_native_attribute(att)
 
-        val = self.get_flux(lsst_bandpasses[band], sed=sed)
+        val = self.get_flux(lsst_bandpasses[band], sed=sed, mjd=mjd)
 
-        if cache:
+        if cache:     # and mjd is None
```

Comment:
I think this line will be executed if `mjd` is set to some number, If `cache == True`, then the subsequent line will raise a `NameError` since `att` will not have been defined.

Reply:
True.   For now I will add the extra condition.